go/simple/v2/ampli/ampli.go

```diff
@@ -0,0 +1,388 @@
+		RequiredNumber: requiredNumber,
+		optionals:      map[string]bool{},
+	}
+}
+
+func (stronglyTypedEvent *Identify) SetOptionalArray(optionalArray []string) *Identify {
+	stronglyTypedEvent.OptionalArray = optionalArray
+	stronglyTypedEvent.optionals["OptionalArray"] = true
+
+	return stronglyTypedEvent
+}
+
+func (stronglyTypedEvent *Identify) ToEvent() amplitude.Event {
+	identify := amplitude.Identify{}
+	identify.Set("requiredNumber", stronglyTypedEvent.RequiredNumber)
+
+	if _, present := stronglyTypedEvent.optionals["OptionalArray"]; present {
+		identify.Set("optionalArray", stronglyTypedEvent.OptionalArray)
+	}
+
+	return amplitude.Event{
+		EventType:      amplitude.IdentifyEventType,
+		UserProperties: identify.Properties,
+	}
+}
+
+// Group
+//
+// [View in Tracking Plan]: https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest/Group
+//
+// Group properties
+//
+// Params:
+//	- RequiredBool: description for group RequiredBool
+//	- OptionalString: description for group OptionalString
+type Group struct {
+	RequiredBoolean bool
+	OptionalString  string
+	groups          map[string][]string
+	optionals       map[string]bool
+}
+
+func NewGroup(requiredBoolean bool) *Group {
+	return &Group{
+		RequiredBoolean: requiredBoolean,
+		optionals:       map[string]bool{},
+	}
+}
+
+func (stronglyTypedEvent *Group) SetOptionalString(optionString string) *Group {
+	stronglyTypedEvent.OptionalString = optionString
+	stronglyTypedEvent.optionals["OptionalString"] = true
+
+	return stronglyTypedEvent
+}
+
+func (stronglyTypedEvent *Group) ToEvent() amplitude.Event {
+	identify := amplitude.Identify{}
+	identify.Set("requiredBoolean", stronglyTypedEvent.RequiredBoolean)
+
+	if _, present := stronglyTypedEvent.optionals["OptionalArray"]; present {
+		identify.Set("optionalString", stronglyTypedEvent.OptionalString)
+	}
+
+	return amplitude.Event{
+		EventType:       amplitude.GroupIdentifyEventType,
+		Groups:          stronglyTypedEvent.groups,
+		GroupProperties: identify.Properties,
+	}
+}
+
+// EventWithAllProperties
+//
+// [View in Tracking Plan]: https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest/Event%20With%20Array%20Types
+//
+// Event with all properties
+//
+// Owner: Test codegen
+//
+// Params
+//	- required_array: Event 2 Property - Array
+//	- required_boolean: Event 2 Property - Boolean
+//	- required_enum: Event 2 Property - Enum
+//	- required_integer: Event 2 Property - Integer
+//	- required_number: Event 2 Property - Number
+//	- required_string: Event 2 Property - String
+//	- optional_string: Event 2 Property - Optional String
+type EventWithAllProperties struct {
+	RequiredArray   []string
+	RequiredBool    bool
+	RequiredEnum    RequiredEnum
+	RequiredInteger int
+	RequiredNumber  float64
+	RequiredString  string
+	OptionalString  string
+	OptionalBool    bool
+	optionals       map[string]bool
+}
+
+type RequiredEnum string
+
+const (
+	RequiredEnumEnum1 RequiredEnum = "Enum1"
```

Comment:
`EventWithAllPropertiesRequiredEnumEnum1`

Reply:
It prevents from potential name conflict. 
Updated.